### ISSUE TITLE
comma dangle functions: ignore throws false positives for all comma dangles

### ISSUE THREAD

**Issue by mnbroatch** [eslint/eslint#7572]
* **ESLint Version:**
3.9.1
* **Node Version:**
6.2.1
* **npm Version:**
3.10.8

**What parser (default, Babel-ESLint, etc.) are you using?**
unsure

```
{
  "extends": [
    "eslint:recommended",
    "plugin:import/errors",
    "plugin:import/warnings",
    "airbnb"
  ],
  "plugins": [
    "react"
  ],
  "parserOptions": {
    "ecmaVersion": 6,
    "sourceType": "module",
    "ecmaFeatures": {
      "jsx": true
    }
  },
  "env": {
    "es6": true,
    "browser": true,
    "node": true,
    "jquery": true,
    "mocha": true
  },
  "rules": {
    "import/no-extraneous-dependencies": [2, { "devDependencies": true } ],
    "no-underscore-dangle":["error",{"allow": ["_id"]}],
    "new-cap": ["error", {"capIsNewExceptions":["Router"]}],
    "no-use-before-define": ["error", { "functions": false, "classes": true }],
    "no-plusplus": ["error", { "allowForLoopAfterthoughts": true }],
    "comma-dangle": ["error", { "functions": "ignore" }]
  }
}




```

**What did you do? Please include the actual source code causing the issue.**

There are many many examples. Here is one:

```js

class TabDisplayArea extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      currentTab: 'Match State',
    };
    this.changeTab = this.changeTab.bind(this);
  }

```

**What did you expect to happen?**

I expect the above to pass.

**What actually happened? Please include the actual, raw output from ESLint.**

many many unexpected trailing commas

Users/matthewbroatch/Programming/Work/demo-with-tabbed-data/src/Reducers/DataReducer.js (15/0)
  ✖   8:37  Unexpected trailing comma  comma-dangle
  ✖  13:23  Unexpected trailing comma  comma-dangle
  ✖  18:29  Unexpected trailing comma  comma-dangle
  ✖  22:29  Unexpected trailing comma  comma-dangle
  ✖  26:29  Unexpected trailing comma  comma-dangle
  ✖  30:29  Unexpected trailing comma  comma-dangle
  ✖  34:29  Unexpected trailing comma  comma-dangle
  ✖  35:8   Unexpected trailing comma  comma-dangle
  ✖  36:6   Unexpected trailing comma  comma-dangle
  ✖  40:29  Unexpected trailing comma  comma-dangle
  ✖  47:38  Unexpected trailing comma  comma-dangle
  ✖  53:38  Unexpected trailing comma  comma-dangle
  ✖  54:6   Unexpected trailing comma  comma-dangle
  ✖  55:4   Unexpected trailing comma  comma-dangle
  ✖  85:51  Unexpected trailing comma  comma-dangle

/Users/matthewbroatch/Programming/Work/demo-with-tabbed-data/src/Reducers/index.js (1/0)
  ✖  5:7  Unexpected trailing comma  comma-dangle

/Users/matthewbroatch/Programming/Work/demo-with-tabbed-data/tools/db/Thing.js (1/0)
  ✖  6:15  Unexpected trailing comma  comma-dangle

/Users/matthewbroatch/Programming/Work/demo-with-tabbed-data/tools/server.js (1/0)
  ✖  33:43  Unexpected trailing comma  comma-dangle

**Comment by not-an-aardvark** [comment]
Hi, thanks for the report.

I'm having some trouble understanding what the issue is, since this looks like intended behavior to me. For example, your `this.state` object unexpectedly has a trailing comma, so it seems like it's correct for the rule to report it.

Could you clarify what behavior you were trying to achieve with `comma-dangle`?

**Comment by mnbroatch** [comment]
You know what? I fixed it. I misunderstood presets.

airbnb's preset overwrites the normal rules for comma dangle, and I didn't realize that by specifying an option for functions in my .eslintrc, it would change the rest of the options to their eslint defaults rather than keep the airbnb settings.

I fixed it by manually setting all options (functions, objects, etc.) for that rule.

Thanks!

**Comment by kaicataldo** [comment]
Glad you resolved your issue!
